src/plugins/usage_collection/server/report/schema.ts

```diff
@@ -21,7 +21,7 @@ import { schema, TypeOf } from '@kbn/config-schema';
 import { METRIC_TYPE } from '@kbn/analytics';
 
 export const reportSchema = schema.object({
-  reportVersion: schema.maybe(schema.literal(1)),
+  reportVersion: schema.maybe(schema.oneOf([schema.literal(1), schema.literal(2)])),
```

Comment:
We report every 30 seconds or a minute so we'll be only losing `uiStatsMetrics ` for the past 30 seconds if the report was not sent during a kibana system upgrade.

Its a very unlikely scenario so i'd rather drop it that complicate the implementation and worry about it in the code

`reportVersion: 1` is still supported since we can make use of user agent and application usage data sent from those reports.